Tasks/DotNetCoreCLIV2/pushcommand.ts

```diff
@@ -182,3 +190,57 @@ function dotNetNuGetPushAsync(dotnetPath: string, packageFile: string, feedUri:
     const envWithProxy = ngRunner.setNuGetProxyEnvironment(process.env, /*configFile*/ null, feedUri);
     return dotnet.exec({ cwd: workingDirectory, env: envWithProxy } as IExecOptions);
 }
+
+function getAccessToken(isInternalFeed: boolean): string{
+    let accessToken: string;
+    let allowServiceConnection = tl.getVariable('PUBLISH_VIA_SERVICE_CONNECTION');
+
+    if(allowServiceConnection) {
+        let endpoint = tl.getInput('externalEndpoint', false);
+
+        if(endpoint && isInternalFeed === true) {
+            tl.debug("Found external endpoint, will use token for auth");
+            let endpointAuth = tl.getEndpointAuthorization(endpoint, true);
+            let endpointScheme = tl.getEndpointAuthorizationScheme(endpoint, true).toLowerCase();
+            switch(endpointScheme)
+            {
+                case ("token"):
+                    accessToken = endpointAuth.parameters["apitoken"];
+                    break;
+                default:
+                    tl.warning("Invalid authentication type for internal feed. Use token based authentication.");
```

Comment:
This message is unclear. I'm not sure what I'd do as a customer in response to it.

Also, messages intended for the user need to be localized. Add an entry to 'messages' in task.json and reference it with `tl.loc`.

Reply:
Gotcha, I swapped it out with wording similar to the create service connection prompts:
**The service connection does not use a supported authentication method. Please use a service connection with personal access token based auth**